core/src/main/java/io/substrait/expression/ExpressionCreator.java

```diff
@@ -284,20 +285,25 @@ public static Expression.ScalarFunctionInvocation scalarFunction(
       SimpleExtension.ScalarFunctionVariant declaration,
       Type outputType,
       FunctionArg... arguments) {
-    return Expression.ScalarFunctionInvocation.builder()
-        .declaration(declaration)
-        .outputType(outputType)
-        .addArguments(arguments)
-        .build();
+      return scalarFunction(declaration, outputType, Arrays.asList(), Arrays.asList(arguments));
+  }
+
+  public static Expression.ScalarFunctionInvocation scalarFunction(
+      SimpleExtension.ScalarFunctionVariant declaration,
+      Type outputType,
+      Iterable<? extends FunctionArg> arguments) {
+    return scalarFunction(declaration, outputType, Arrays.asList(), arguments);
   }
 
   public static Expression.ScalarFunctionInvocation scalarFunction(
       SimpleExtension.ScalarFunctionVariant declaration,
       Type outputType,
+      List<? extends FunctionOption> options,
```

Comment:
I wasn't sure if it's better to change the existing functions or add new ones given this is an optional parameter, so I added new ones. But happy to remove the old ones too if that'd be better?

Reply:
I think were getting close to the point where this pattern (individual constructor variants with different field combinations) may not be good as we start to explode the number of function combinations. These are fine for now, but if were to add more beyond these it might make sense to revisit this utility function approach.

Reply:
That's fair - I decided to revert the additions and add some docstrings to point to using the builder directly in https://github.com/substrait-io/substrait-java/pull/278/commits/9aae3ec7461e9363997d5f7e856bd290b91c9668, does that seem alright to you?